lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java

```diff
@@ -162,56 +155,67 @@ public Evaluation get(String id) throws DatastoreException, NotFoundException {
 		MapSqlParameterSource param = new MapSqlParameterSource();
+		param.addValue(CONTENT_SOURCE, KeyFactory.stringToKey(id));
+		// parameters here are 
+		//	BIND_VAR_PREFIX, appended with 0,1,2,... (to bind group id)
+		//	AuthorizationSqlUtil.ACCESS_TYPE_BIND_VAR (to bind ACCESS_TYPE, e.g. 'SUBMIT')
+		for (int i=0; i<principalIds.size(); i++) {
+			param.addValue(BIND_VAR_PREFIX+i, principalIds.get(i));	
+		}
+		param.addValue(ACCESS_TYPE_BIND_VAR, accessType.name());
 		param.addValue(OFFSET_PARAM_NAME, offset);
 		param.addValue(LIMIT_PARAM_NAME, limit);	
-		List<EvaluationDBO> dbos = namedJdbcTemplate.query(SELECT_ALL_SQL_PAGINATED, param, rowMapper);
+		param.addValue(RESOURCE_TYPE_BIND_VAR, ObjectType.EVALUATION.name());
+		StringBuilder sql = new StringBuilder(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_PREFIX);
+		sql.append(AuthorizationSqlUtil.authorizationSQLWhere(principalIds.size(), 0));
+		sql.append(SELECT_AVAILABLE_CONTENT_SOURCE_FILTER);
+		sql.append(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_SUFFIX);
+
+		List<EvaluationDBO> dbos = namedJdbcTemplate.query(sql.toString(), param, rowMapper);
 		List<Evaluation> dtos = new ArrayList<Evaluation>();
 		copyDbosToDtos(dbos, dtos);
 		return dtos;
 	}
-
+	
+	/**
+	 * return the evaluations in which the user (given as a list of principal Ids)
+	 * has the given access type
+	 */
 	@Override
-	public List<Evaluation> getInRange(long limit, long offset,
-			EvaluationStatus status) throws DatastoreException,
-			NotFoundException {
+	public List<Evaluation> getAvailableInRange(List<Long> principalIds, ACCESS_TYPE accessType, long limit, long offset, List<Long> evaluationIds) throws DatastoreException {
+		if (principalIds.isEmpty()) return new ArrayList<Evaluation>(); // SQL breaks down if list is empty
 		MapSqlParameterSource param = new MapSqlParameterSource();
+		// parameters here are 
+		//	BIND_VAR_PREFIX, appended with 0,1,2,... (to bind group id)
+		//	AuthorizationSqlUtil.ACCESS_TYPE_BIND_VAR (to bind ACCESS_TYPE, e.g. 'SUBMIT')
+		for (int i=0; i<principalIds.size(); i++) {
+			param.addValue(BIND_VAR_PREFIX+i, principalIds.get(i));	
+		}
+		param.addValue(ACCESS_TYPE_BIND_VAR, accessType.name());
 		param.addValue(OFFSET_PARAM_NAME, offset);
 		param.addValue(LIMIT_PARAM_NAME, limit);	
-		param.addValue(STATUS, status.ordinal());
-		List<EvaluationDBO> dbos = namedJdbcTemplate.query(SELECT_BY_STATUS_SQL_PAGINATED, param, rowMapper);
+		param.addValue(RESOURCE_TYPE_BIND_VAR, ObjectType.EVALUATION.name());
+		StringBuilder sql = new StringBuilder(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_PREFIX);
+		sql.append(AuthorizationSqlUtil.authorizationSQLWhere(principalIds.size(), 0));
+		if (evaluationIds==null || evaluationIds.isEmpty()) {
+			sql.append(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_SUFFIX);
+		} else {
+			param.addValue(COL_EVALUATION_ID, evaluationIds);
+			sql.append(SELECT_AVAILABLE_EVALUATIONS_FILTER);
+			sql.append(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_SUFFIX);
```

Comment:
factor out of the if/else this line "sql.append(SELECT_AVAILABLE_EVALUATIONS_PAGINATED_SUFFIX);"